### ISSUE TITLE
Typst table alignment depends on `echo: true/false`

### ISSUE THREAD

**Issue by mvuorre** [quarto-dev/quarto-cli#10086]
### Bug description

I am writing a basic Quarto document with vanilla Typst format:

````
---
title: "Hello, Quarto"
format: 
  typst: default
---

```{r}
#| label: tbl-example1
#| tbl-cap: Table created with `knitr::kable()`
#| echo: true

set.seed(1)
data <- data.frame(x = rnorm(30), y = rnorm(30))
knitr::kable(
  head(data),
  digits = 2
)
```
````

This produces the output below with the expected centered table:

<img width="886" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/7349270/5e2efb96-4ddd-413f-8822-9bd09aa86a56">

I then change `#| echo: true` to `#| echo: false` below, and get a left-aligned table instead, which seems like a bug or at least unexpected behavior.

````
---
title: "Hello, Quarto"
format: 
  typst: default
---

```{r}
#| label: tbl-example1
#| tbl-cap: Table created with `knitr::kable()`
#| echo: false

set.seed(1)
data <- data.frame(x = rnorm(30), y = rnorm(30))
knitr::kable(
  head(data),
  digits = 2
)
```
````

<img width="892" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/7349270/b51106c8-14e6-4a04-9abd-1bccc75be507">

(Possibly related to #10065)

### Steps to reproduce

To reproduce the unexpected behavior (non-centered table), render this:

````
---
title: "Hello, Quarto"
format: 
  typst: default
---

```{r}
#| label: tbl-example1
#| tbl-cap: Table created with `knitr::kable()`
#| echo: false

set.seed(1)
data <- data.frame(x = rnorm(30), y = rnorm(30))
knitr::kable(
  head(data),
  digits = 2
)
```
````

### Expected behavior

I should get a center-aligned table.

### Actual behavior

I instead get a left-aligned table.

### Your environment

- IDE: RStudio 2024.07.0 Build 167
- OS: MacOS 14.5

### Quarto check output

Quarto 1.5.46
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.5.46
      Path: /Applications/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: Installation From Path
      Path: /Library/TeX/texbin
      Version: 2024

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.12.3
      Path: /opt/homebrew/opt/python@3.12/bin/python3.12
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with python3 -m pip install jupyter

[✓] Checking R installation...........OK
      Version: 4.4.1
      Path: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources
      LibPaths:
        - /Users/matti/Library/R/arm64/4.4/library
        - /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/library
      knitr: 1.46
      rmarkdown: 2.27

[✓] Checking Knitr engine render......OK

**Comment by cderv** [comment]
I can reproduce this. 

<table>
<tr>
<td> 

`echo: false`

<td> 

`echo: true`
<tr>
<td>

````markdown
::: {#tbl-example1 .cell tbl-cap='Table created with `knitr::kable()`'}
::: {.cell-output-display}


|     x|     y|
|-----:|-----:|
| -0.63|  1.36|
|  0.18| -0.10|
| -0.84|  0.39|
|  1.60| -0.05|
|  0.33| -1.38|
| -0.82| -0.41|


:::
:::
````

<td>

````markdown
::: {#tbl-example1 .cell tbl-cap='Table created with `knitr::kable()`'}

```{.r .cell-code}
set.seed(1)
data <- data.frame(x = rnorm(30), y = rnorm(30))
knitr::kable(
  head(data),
  digits = 2
)
```

::: {.cell-output-display}


|     x|     y|
|-----:|-----:|
| -0.63|  1.36|
|  0.18| -0.10|
| -0.84|  0.39|
|  1.60| -0.05|
|  0.33| -1.38|
| -0.82| -0.41|


:::
:::
````

</table>

and we generate this 

<table>
<tr>
<td> 

`echo: false`

<td> 

`echo: true`
<tr>
<td>

````typ
#block[
```r
set.seed(1)
data <- data.frame(x = rnorm(30), y = rnorm(30))
knitr::kable(
  head(data),
  digits = 2
)
```

#figure([
#block[
#figure(
  align(center)[#table(
    columns: 2,
    align: (right,right,),
    table.header([x], [y],),
    table.hline(),
    [-0.63], [1.36],
    [0.18], [-0.10],
    [-0.84], [0.39],
    [1.60], [-0.05],
    [0.33], [-1.38],
    [-0.82], [-0.41],
  )]
  , kind: table
  )

]
], caption: figure.caption(
position: top, 
[
Table created with `knitr::kable()`
]), 
kind: "quarto-float-tbl", 
supplement: "Table", 
)
<tbl-example1>

````

<td>

````typ
#block[
#figure([
#block[
#figure(
  align(center)[#table(
    columns: 2,
    align: (right,right,),
    table.header([x], [y],),
    table.hline(),
    [-0.63], [1.36],
    [0.18], [-0.10],
    [-0.84], [0.39],
    [1.60], [-0.05],
    [0.33], [-1.38],
    [-0.82], [-0.41],
  )]
  , kind: table
  )

]
], caption: figure.caption(
position: top, 
[
Table created with `knitr::kable()`
]), 
kind: "quarto-float-tbl", 
supplement: "Table", 
)
<tbl-example1>

````

So we do keep one extra `#block` that may impact the centering - we correctly get `align(center)` in both... 🤔 

@gordonwoodhull do you know typst enough to see the problem right-away ?

**Comment by cscheid** [comment]
We need to either lose that `#block()` call, or add `width: 100%` to it (and all its parents). This is a feature/limitation of typst sizing.

**Comment by cscheid** [comment]
(This feels worth it to fix for 1.5 if we can get it in the next couple of days.)

**Comment by gordonwoodhull** [comment]
Yeah, I solved a few cases where we need to mark with scaffold, but did not thoroughly work through the possibilities.

Is the double `figure` also an issue?

**Comment by cscheid** [comment]
> Is the nested `figure` also an issue?

It is, but unfortunately it's a Pandoc issue:

````
$ pandoc -f markdown -t typst -o -
| foo | bar |
|-----|-----|
| 1   | 2   |
^D
#figure(
  align(center)[#table(
    columns: 2,
    align: (auto,auto,),
    table.header([foo], [bar],),
    table.hline(),
    [1], [2],
  )]
  , kind: table
  )
````

**Comment by gordonwoodhull** [comment]
This is another instance of #8797, partly solved in #9052

We `quarto.utils.match` patterns over the AST in `quarto-post/typst.lua` and add the scaffold class. Need to refine / add more patterns.
